<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Software Tutorial</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fusiomr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="software_tutorial.html">Software Tutorial</a>
</li>
<li>
  <a href="data_analysis.html">Data Analysis</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Software Tutorial</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>fusiomr/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250723code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250723)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250723code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250723)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkangbw702FusioMRtreeae124caca5910e6c9a6c46c8663f55ec7767323etargetblankae124caa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/kangbw702/FusioMR/tree/ae124caca5910e6c9a6c46c8663f55ec7767323e" target="_blank">ae124ca</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomkangbw702FusioMRtreeae124caca5910e6c9a6c46c8663f55ec7767323etargetblankae124caa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/kangbw702/FusioMR/tree/ae124caca5910e6c9a6c46c8663f55ec7767323e" target="_blank">ae124ca</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    src/gibbs_memo_joint.o

Unstaged changes:
    Modified:   README.md
    Modified:   fusiomr/analysis/_site.yml
    Modified:   fusiomr/analysis/data_analysis.Rmd
    Modified:   fusiomr/analysis/index.Rmd
    Modified:   fusiomr/analysis/software_tutorial.Rmd
    Deleted:    tutorial/.Rprofile
    Deleted:    tutorial/.gitattributes
    Deleted:    tutorial/.gitignore
    Deleted:    tutorial/README.md
    Deleted:    tutorial/_workflowr.yml
    Deleted:    tutorial/analysis/_site.yml
    Deleted:    tutorial/analysis/about.Rmd
    Deleted:    tutorial/analysis/index.Rmd
    Deleted:    tutorial/analysis/license.Rmd
    Deleted:    tutorial/code/README.md
    Deleted:    tutorial/data/README.md
    Deleted:    tutorial/docs/.nojekyll
    Deleted:    tutorial/output/README.md
    Deleted:    tutorial/tutorial.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>fusiomr/analysis/software_tutorial.Rmd</code>) and HTML
(<code>fusiomr/docs/software_tutorial.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kangbw702/FusioMR/blob/ae124caca5910e6c9a6c46c8663f55ec7767323e/fusiomr/analysis/software_tutorial.Rmd" target="_blank">ae124ca</a>
</td>
<td>
Sihao
</td>
<td>
2025-07-23
</td>
<td>
update tutorial
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kangbw702/FusioMR/ae124caca5910e6c9a6c46c8663f55ec7767323e/fusiomr/docs/software_tutorial.html" target="_blank">ae124ca</a>
</td>
<td>
Sihao
</td>
<td>
2025-07-23
</td>
<td>
update tutorial
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>With provided genome-wide summary statistics (IV-exposure effect
estimates, IV-outcome effect estimates, standard errors for IV-exposure
effect, standard errors for IV-outcome effect), FusioMR returns a
estimated causal effect between gene and trait.</p>
<p>This R package supports four different analytical models:</p>
<ul>
<li><strong>Model 1</strong>: Single Exposure, Single Outcome, No
Correlated Horizontal Pleiotropy</li>
<li><strong>Model 2</strong>: Single Exposure, Single Outcome, With
Correlated Horizontal Pleiotropy</li>
<li><strong>Model 3</strong>: Single Exposure, Multiple Outcomes, No
Correlated Horizontal Pleiotropy</li>
<li><strong>Model 4</strong>: Multiple Exposure, Multiple Outcomes, With
both Uncorrelated and Correlated Horizontal Pleiotropy</li>
</ul>
<div id="main-function" class="section level2">
<h2>Main Function</h2>
<p><code>fusiomr()</code></p>
<p><strong>Purpose</strong>: Perform causal effect estimation with
provided GWAS summary statistics.</p>
<p><strong>Arguments</strong>:</p>
<table>
<colgroup>
<col width="28%" />
<col width="15%" />
<col width="33%" />
<col width="23%" />
</colgroup>
<thead>
<tr class="header">
<th>Parameter</th>
<th>Type</th>
<th>Description</th>
<th>Default</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>b_exp</code></td>
<td>numeric vector/matrix</td>
<td>IV-exposure effect estimates</td>
<td>required</td>
</tr>
<tr class="even">
<td><code>se_exp</code></td>
<td>numeric vector/matrix</td>
<td>IV-exposure standard errors</td>
<td>required</td>
</tr>
<tr class="odd">
<td><code>b_out</code></td>
<td>numeric vector/matrix</td>
<td>IV-outcome effect estimates</td>
<td>required</td>
</tr>
<tr class="even">
<td><code>se_out</code></td>
<td>numeric vector/matrix</td>
<td>IV-outcome standard errors</td>
<td>required</td>
</tr>
<tr class="odd">
<td><code>CHP</code></td>
<td>logical</td>
<td>Include horizontal pleiotropy correction</td>
<td><code>FALSE</code></td>
</tr>
<tr class="even">
<td><code>p_value_threshold</code></td>
<td>numeric</td>
<td>IV selection p-value threshold</td>
<td><code>1e-3</code></td>
</tr>
<tr class="odd">
<td><code>niter</code></td>
<td>integer</td>
<td>Number of Gibbs sampling iterations</td>
<td><code>20000</code></td>
</tr>
<tr class="even">
<td><code>burnin_prop</code></td>
<td>numeric</td>
<td>Proportion of iterations to discard as burn-in</td>
<td><code>0.5</code></td>
</tr>
</tbody>
</table>
<p><strong>Note</strong>: For multiple exposure, multiple outcome
models, the CHP parameter is automatically set to <code>TRUE</code>
regardless of user input. This is because the model is specifically
designed for complex trait exposures where both correlated horizontal
pleiotropy (CHP) and uncorrelated horizontal pleiotropy (UHP) effects
are expected and need to be accounted for by default.</p>
<p><strong>Returns</strong>: The <code>fusiomr()</code> function returns
a list containing:</p>
<ul>
<li><code>est</code>: Posterior mean(s) of causal effect(s)</li>
<li><code>se</code>: Posterior standard error(s)</li>
<li><code>pval</code>: P-value(s)</li>
</ul>
</div>
<div id="input-data-requirement" class="section level2">
<h2>Input Data Requirement</h2>
<ul>
<li><code>b_exp</code>: Must have exactly 1 column and same number of
rows as other inputs.<br />
</li>
<li><code>se_exp</code>: Must have exactly 1 column, same dimensions as
b_exp, and all positive values.<br />
</li>
<li><code>b_out</code>: Must have 1 column (single outcome) or 2 columns
(multiple outcomes), with same number of rows as b_exp.<br />
</li>
<li><code>se_out</code>: Must have same dimensions as b_out and all
positive values.<br />
</li>
<li><code>CHP</code>: Either FALSE or TRUE.<br />
</li>
<li><code>p_value_threshold</code>: Must be positive.<br />
</li>
<li><code>niter</code>: Must be positive integer specifying number of
Gibbs sampling iterations.<br />
</li>
<li><code>burnin_prop</code>:Must be numeric value between 0 and 1.</li>
</ul>
</div>
</div>
<div id="simple-example-usage" class="section level1">
<h1>Simple Example Usage</h1>
<div id="step1-install-fusiomr" class="section level2">
<h2>STEP1: Install FusioMR</h2>
<p>You can install the development version of FusioMR from GitHub
with:</p>
<pre class="r"><code># Install devtools if you don&#39;t have it
install.packages(&quot;devtools&quot;)

# Install FusioMR from your GitHub repository
devtools::install_github(&quot;kangbw702/FusioMR&quot;)</code></pre>
</div>
<div id="step2-run-estimation" class="section level2">
<h2>STEP2: Run Estimation</h2>
<!--edit-->
<div id="single-exposure-single-outcome-analysis"
class="section level3">
<h3>Single Exposure Single Outcome Analysis</h3>
<p>Simply, with required summary statistics, you can run estimation for
single exposure single outcome as below:</p>
<pre class="r"><code>library(FusioMR)

# Prepare your data
b_exp &lt;- c(0.12, 0.08, 0.15, 0.09)    # IV-exposure effects
se_exp &lt;- c(0.01, 0.01, 0.02, 0.01)   # IV-exposure standard errors
b_out &lt;- c(0.05, 0.03, 0.07, 0.04)    # IV-outcome effects  
se_out &lt;- c(0.02, 0.02, 0.03, 0.02)   # IV-outcome standard errors

# Run single outcome analysis
result &lt;- fusiomr(
  b_exp = b_exp,
  se_exp = se_exp, 
  b_out = b_out,
  se_out = se_out
)

# View results
print(result)</code></pre>
<p>Running this, you are expected to get the estimation results in
similar format:</p>
<pre class="r"><code>#&gt; $est
#&gt; [1] 0.4333415
#&gt; 
#&gt; $se  
#&gt; [1] 0.1003104
#&gt; 
#&gt; $pval
#&gt; [1] 1.56024e-05</code></pre>
<p>If you would like to run the analysis with correlated horizontal
pleiotropy, you can enter customized value for the function
parameters:</p>
<pre class="r"><code>result &lt;- fusiomr(
  b_exp = b_exp, 
  se_exp = se_exp,
  b_out = b_out, 
  se_out = se_out,
  CHP = TRUE,              # Include correlated horizontal pleiotropy
  niter = 50000,           # More iterations
  burnin_prop = 0.3,       # Shorter burn-in
  p_value_threshold = 5e-8 # More stringent threshold
)</code></pre>
</div>
<div id="single-exposure-multiple-outcome-analysis"
class="section level3">
<h3>Single Exposure Multiple Outcome Analysis</h3>
<p>Simply, with required summary statistics, you can run estimation for
single exposure multiple outcomes as below:</p>
<pre class="r"><code>library(FusioMR)

# Prepare your input data
b_exp &lt;- c(0.12, 0.08, 0.15, 0.09)    # IV-exposure effects
se_exp &lt;- c(0.01, 0.01, 0.02, 0.01)   # IV-exposure standard errors
b_out_multi &lt;- cbind(                 # IV-outcome effects 
  outcome1 = c(0.05, 0.03, 0.07, 0.04),
  outcome2 = c(0.08, 0.06, 0.11, 0.07)
)
se_out_multi &lt;- cbind(                # IV-outcome standard errors 
  outcome1 = c(0.02, 0.02, 0.03, 0.02),
  outcome2 = c(0.025, 0.02, 0.03, 0.025)
)

# Run multiple outcomes analysis
result_multi &lt;- fusiomr(
  b_exp = b_exp,
  se_exp = se_exp,
  b_out = b_out_multi,
  se_out = se_out_multi
)</code></pre>
<p>Running this, you are expected to get the estimation results in
similar format:</p>
<pre class="r"><code>#&gt; $est
#&gt; [1] 0.4466227 0.7525042
#&gt; 
#&gt; $se
#&gt; [1] 0.1798687 0.1836945
#&gt; 
#&gt; $pval
#&gt; [1] 1.298748e-02 4.131123e-05</code></pre>
</div>
<div id="multiple-exposure-multiple-outcome-analysis"
class="section level3">
<h3>Multiple Exposure Multiple Outcome Analysis</h3>
<p>Simply, with required summary statistics, you can run estimation for
multiple exposure multiple outcomes as below:</p>
<pre class="r"><code>library(FusioMR)

# Prepare your input data
b_exp_multi &lt;- cbind(
  exposure1 = c(0.12, 0.08, 0.15, 0.09),
  exposure2 = c(0.14, 0.06, 0.17, 0.09)
)
se_exp_multi &lt;- cbind(
  exposure1 = c(0.01, 0.01, 0.02, 0.01),
  exposure2 = c(0.025, 0.01, 0.03, 0.075)
)
b_out_multi &lt;- cbind(                
  outcome1 = c(0.05, 0.03, 0.07, 0.04),
  outcome2 = c(0.08, 0.06, 0.11, 0.07)
)
se_out_multi &lt;- cbind(               
  outcome1 = c(0.02, 0.02, 0.03, 0.02),
  outcome2 = c(0.025, 0.02, 0.03, 0.025)
)

# Run multiple outcomes analysis
result_multi &lt;- fusiomr(
  b_exp = b_exp_multi,
  se_exp = se_exp_multi,
  b_out = b_out_multi,
  se_out = se_out_multi
)</code></pre>
<p>Running this, you are expected to get the estimation results in
similar format:</p>
<pre class="r"><code># View results for both outcomes
print(result_multi)
#&gt;$est
#&gt;[1] 0.4340691 0.7533179
#&gt;
#&gt;$se
#&gt;[1] 0.1081232 0.4509967
#&gt;
#&gt;$pval
#&gt;[1] 5.955244e-05 9.485208e-02</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.3 (2025-02-28)
Platform: x86_64-apple-darwin20
Running under: macOS Sequoia 15.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       cli_3.6.5         knitr_1.50        rlang_1.1.6      
 [5] xfun_0.52         stringi_1.8.7     promises_1.3.2    jsonlite_2.0.0   
 [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.36.2     
[13] htmltools_0.5.8.1 httpuv_1.6.16     sass_0.4.10       rmarkdown_2.29   
[17] evaluate_1.0.3    jquerylib_0.1.4   tibble_3.2.1      fastmap_1.2.0    
[21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
[25] compiler_4.4.3    fs_1.6.6          Rcpp_1.1.0        pkgconfig_2.0.3  
[29] rstudioapi_0.17.1 later_1.4.2       digest_0.6.37     R6_2.6.1         
[33] pillar_1.10.2     magrittr_2.0.3    bslib_0.9.0       tools_4.4.3      
[37] cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
